test/e2e/storage/testsuites/volumemode.go

```diff
@@ -182,27 +183,47 @@ func (t *volumeModeTestSuite) defineTests(driver TestDriver, pattern testpattern
 
 				ginkgo.By("Creating sc")
 				l.sc, err = l.cs.StorageV1().StorageClasses().Create(l.sc)
-				framework.ExpectNoError(err)
+				framework.ExpectNoError(err, "Failed to create sc")
 
 				ginkgo.By("Creating pv and pvc")
 				l.pv, err = l.cs.CoreV1().PersistentVolumes().Create(l.pv)
-				framework.ExpectNoError(err)
+				framework.ExpectNoError(err, "Failed to create pv")
 
 				// Prebind pv
 				l.pvc.Spec.VolumeName = l.pv.Name
 				l.pvc, err = l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Create(l.pvc)
-				framework.ExpectNoError(err)
+				framework.ExpectNoError(err, "Failed to create pvc")
 
-				framework.ExpectNoError(framework.WaitOnPVandPVC(l.cs, l.ns.Name, l.pv, l.pvc))
+				framework.ExpectNoError(framework.WaitOnPVandPVC(l.cs, l.ns.Name, l.pv, l.pvc), "Failed to bind pv and pvc")
 
 				ginkgo.By("Creating pod")
-				pod, err := framework.CreateSecPodWithNodeSelection(l.cs, l.ns.Name, []*v1.PersistentVolumeClaim{l.pvc},
-					nil, false, "", false, false, framework.SELinuxLabel,
-					nil, framework.NodeSelection{Name: l.config.ClientNodeName}, framework.PodStartTimeout)
+				pod := framework.MakeSecPod(l.ns.Name, []*v1.PersistentVolumeClaim{l.pvc}, nil, false, "", false, false, framework.SELinuxLabel, nil)
+				// Setting node
+				pod.Spec.NodeName = l.config.ClientNodeName
+				pod, err = l.cs.CoreV1().Pods(l.ns.Name).Create(pod)
 				defer func() {
-					framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod))
+					framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod), "Failed to delete pod")
 				}()
-				framework.ExpectError(err)
+				framework.ExpectNoError(err, "Failed to create pod")
```

Comment:
If an error happens here, I guess the above `defer func() {` causes another error because of nonexistent pod.

Reply:
@oomichi

Thank you for your review.

`DeletePodWithWait` handles `pod == nil` case properly, so no additional error will happen.
However, I've moved `ExpectNoError` before `defer` for readability, in the second commit.

PTAL

(BTW, original code was intentionaly implemented with that order, because it expects such a case that pod creation succeeds but it doesn't become running state, where pod needs to be deleted. By this PR, we won't need to care about it, for we check them separatedly.)